splice/queries/distribution.py

```diff
@@ -154,3 +164,48 @@ def get_possible_distributions(today=None, channel_id=None):
             "force_upload": True
         })
     return artifacts
+
+
+_SENTINEL = object()
+
+
+def multiplex_directory_tiles(tiles):
+    """Directory tile multiplexer that creates all possible combinations of the
+    tile sets based on its type and the target url. Given multiple sponsored tiles,
+    it'll pick one of them and create a new tile set with other directory tiles.
+    It also multiplexes the tiles with the same TLD+1 target url (e.g. www.mozilla.org).
```

Comment:
it's "TLD+1", we only want to multiplex those tiles with the same "TLD+1", for example, "www.mozilla.org/pocket", "www.mozilla.org/principles", ..., "www.mozilla.org/gears". Both of them have "TLD+1" as "www.mozilla.org"